core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -314,10 +309,13 @@ private void finishAsFailed(Exception exception) {
         }
     }
 
+    /**
+     * An encapsulation of an operation that is to be performed on the primary shard
+     */
     public interface Primary<
-                Request extends ReplicationRequest<Request>,
-                ReplicaRequest extends ReplicationRequest<ReplicaRequest>,
-                PrimaryResultT extends PrimaryResult<ReplicaRequest>
+                R extends ReplicationRequest<R>,
```

Comment:
I have seen two standard "formats", one is the single char format, which the JDK uses a lot like `Collection<E>`, so that's why the `R`. For the others, `PrimaryResult` as `PR` I could do, but I figured since it wasn't a single char I'd suffix it with `T` for type (another format I've seen in codebases).

I'm happy to change it to `R`, and `PR` if you think that would be clearer